src/Utilities/ContainerHelpers.hpp

```diff
@@ -122,3 +123,50 @@ SPECTRE_ALWAYS_INLINE decltype(auto) get_size(
       tt::is_complex_of_fundamental_v<std::remove_cv_t<T>> or
       cpp17::is_fundamental_v<std::remove_cv_t<T>>)>::get_size(t, size);
 }
+
+namespace ContainerHelpers_detail {
+template <typename T>
+auto get_used_for_size_impl(
+    const size_t size, T /*meta*/,
+    cpp17::bool_constant<false> /*is fundamental*/,
+    cpp17::bool_constant<true> /*is vectorimpl*/) noexcept {
+  return T(size);
```

Comment:
This does an allocation just to set the size of a container.  When this was just a utility for tests that was fine, but I don't think we want to do this in production code.

Reply:
I ended up adding and additional `MakeWithValue` for vectors, because it turned out to be way easier than I had initially thought and seems more satisfying because then even the tests don't do the extra unneeded allocation. If you don't like it, let me know and I'll just put the `get_used_for_size` in the test helpers.